lightly_studio/src/lightly_studio/vendor/perception_encoder/vision_encoder/config.py

```diff
@@ -7,28 +10,57 @@
 Include all available vision encoder configurations.
 """
 
-from dataclasses import dataclass, replace
+from __future__ import annotations
 
-from functools import partial
-from typing import Callable, Optional, Sequence, Tuple, List
+from dataclasses import dataclass
 
-from huggingface_hub import hf_hub_download
+import os
+from pathlib import Path
+import tempfile
+from typing import Optional
+import urllib
 
+from lightly_studio.dataset import file_utils
 
 
-def fetch_pe_checkpoint(name: str, path: Optional[str] = None):
-    path = path or f"hf://facebook/{name}:{name}.pt"
+DOWNLOADABLE_MODEL_BASE_URL = (
+    "https://lightly-studio-models.s3.us-east-1.amazonaws.com"
+)
+
+DOWNLOADABLE_MODEL_URL: dict[str, str] = {
+    "PE-Core-T16-384": "perception_encoder/core/PE-Core-T16-384.pt",
+    "PE-Core-S16-384": "perception_encoder/core/PE-Core-S16-384.pt",
+    "PE-Core-B16-224": "perception_encoder/core/PE-Core-B16-224.pt",
+    "PE-Core-L14-336": "perception_encoder/core/PE-Core-L14-336.pt",
+    "PE-Core-G14-448": "perception_encoder/core/PE-Core-G14-448.pt",
+}
+
+def download_checkpoint(model_name: str) -> Path:
+    """Downloads a checkpoint and returns the local path to it.
 
-    if path.startswith("hf://"):
-        # Load from huggingface
-        path = path[len("hf://"):]
-        repo, file = path.split(":")
+    Supports checkpoints from:
+    - Predefined downloadable model names from our repository
 
-        return hf_hub_download(repo_id=repo, filename=file)
-    else:
-        return path
+    Returns:
+        Path to the local checkpoint file.
+    """
+    if model_name in DOWNLOADABLE_MODEL_URL.keys():
+        model_url = DOWNLOADABLE_MODEL_URL[model_name]
+        model_url = urllib.parse.urljoin(DOWNLOADABLE_MODEL_BASE_URL, model_url)
 
+        download_dir = Path(tempfile.gettempdir())
+        model_name_with_suffix = os.path.basename(urllib.parse.urlparse(model_url).path)
+        local_ckpt_path = download_dir / model_name_with_suffix
 
+        file_utils.download_file_if_does_not_exist(
+            url=model_url,
+            local_filename=local_ckpt_path,
+        )
+        return str(local_ckpt_path)
```

Comment:
The type annotations says Path. I think it is fine to return str, we don't need the cast in fetch_pe_checkpoint then.

Note that vendored code is not static checked (on purpose).